pkg/endpoint/bpf.go

```diff
@@ -825,7 +825,10 @@ func (e *Endpoint) SkipStateClean() {
 	e.unlock()
 }
 
-func (e *Endpoint) deletePolicyKey(keyToDelete policy.Key, incremental bool) bool {
+// The bool pointed by hadProxy, if not nil, will be set to 'true' if
+// the deleted entry had a proxy port assigned to it.  *hadProxy is
+// not otherwise changed (e.g., it is never set to 'false').
+func (e *Endpoint) deletePolicyKey(keyToDelete policy.Key, incremental bool, hadProxy *bool) bool {
```

Comment:
I'm surprised that `hadProxy` is passed in as a pointer instead of returning it. I'm guessing changing the return means changing the callsites for `deletePolicyKey` in a bigger way? (since the ->true accumulation would have to happen at outside here).

Reply:
Right, it is not strictly a return parameter, and this way the PR become simpler.